src/test/mixxxtest.cpp

```diff
@@ -65,3 +68,79 @@ MixxxTest::~MixxxTest() {
         pControl->deleteCreatorCO();
     }
 }
+
+void MixxxTest::saveAndReloadConfig() {
+    m_pConfig->save();
+    m_pConfig = UserSettingsPointer(
+            new UserSettings(getTestDataDir().filePath("test.cfg")));
+    ControlDoublePrivate::setUserConfig(m_pConfig);
+}
+
+namespace test {
+
+FileRemover::~FileRemover() {
+    VERIFY_OR_DEBUG_ASSERT(
+            m_fileName.isEmpty() ||
+            QFile::remove(m_fileName) ||
+            !QFile::exists(m_fileName)) {
+        // unexpected
+    }
+}
+
+QString generateTemporaryFileName(const QString& fileNameTemplate) {
+    auto tmpFile = QTemporaryFile(fileNameTemplate);
+    // The file must be opened to create it and to obtain
+    // its file name!
+    tmpFile.open();
+    const auto tmpFileName = tmpFile.fileName();
+    DEBUG_ASSERT(!tmpFileName.isEmpty());
+    // The empty temporary file will be removed upon returning
+    // from this function
+    return tmpFileName;
+}
+
+QString createEmptyTemporaryFile(const QString& fileNameTemplate) {
+    const auto fileName = generateTemporaryFileName(fileNameTemplate);
```

Comment:
Wy can't we use QTemporaryFile directly?

Reply:
Because we need an empty existing file.